app/scripts/directives/nav.js

```diff
@@ -342,8 +343,14 @@ angular.module('openshiftConsole')
               $scope.canIAddToProject = AuthorizationService.canIAddToProject(currentProjectName);
 
               if ($scope.canIAddToProject) {
-                CatalogService.getCatalogItems().then(function(items) {
-                  $scope.catalogItems = items;
+                var catalogItemsPromise = CatalogService.getCatalogItems().then(function(items) {
+                  catalogItems = items;
+                });
+                var projectItemsPromise = Catalog.getProjectCatalogItems(currentProjectName).then(_.spread(function(catalogServiceItems) {
+                  projectItems = catalogServiceItems;
+                }));
+                $q.all([catalogItemsPromise, projectItemsPromise]).then(function() {
+                  $scope.catalogItems = Catalog.sortCatalogItems(_.concat(catalogItems, projectItems));
                 });
```

Comment:
What happens if there is a failure getting catalogItems and/or projectItems?   There doesn't seem to be a failure/error block.

Reply:
An error is displayed by data service. Right now catalog service in origin-web-catalog doesn't tell data service to suppress error notifications.